src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -712,7 +721,7 @@ private static DataContract CreateDataContract(int id, RuntimeTypeHandle typeHan
 
                                     if (!CollectionDataContract.TryCreate(type, out dataContract))
```

Comment:
On the full framework, the following code appear just before this. Should it be put back in? The IsPointer property has been around for a while.  
```csharp
if (type.IsPointer)
    type = Globals.TypeOfReflectionPointer;
```